src/Libraries/Microsoft.Extensions.AI.OpenAI/OpenAIRealtimeExtensions.cs

```diff
@@ -107,7 +107,7 @@ public static async Task HandleToolCallsAsync(
 
             try
             {
-                var result = await aiFunction.InvokeAsync(functionCallContent.Arguments, cancellationToken).ConfigureAwait(false);
+                var result = await aiFunction.InvokeAsync(new(functionCallContent.Arguments), cancellationToken).ConfigureAwait(false);
```

Comment:
Could we pass in an `IServiceProvider` here too? I know it would have to be threaded through from `HandleToolCallsAsync` but otherwise it might seem like a missing capability.

Reply:
I agree this feels like a gap. Is this and FunctionInvokingChatClient the only two places we call AIFunction.InvokeAsync internally?